app/services/authentication/providers/apple.rb

```diff
@@ -0,0 +1,78 @@
+module Authentication
+  module Providers
+    # Apple authentication provider, uses omniauth-apple as backend
+    class Apple < Provider
+      SETTINGS_URL = "https://appleid.apple.com/account/manage".freeze
+      TRUSTED_CALLBACK_ORIGIN = "https://appleid.apple.com".freeze
+      CALLBACK_PATH = "/users/auth/apple/callback".freeze
+
+      # Currently `beta` state until thoroughly tested in production level env
+      def self.beta_access?
+        true
```

Comment:
If we wanted to make this more dynamic we could hook this value up to a feature flag (totally optional though)

Reply:
Sounds like a good idea. I'm trying it out locally, it does give more flexibility to this beta restriction.

The feature flag if enabled would make the auth provider publicly available (act as if not beta) but otherwise would be hidden behind the param `state=beta_providers_enabled`. Forem instances would always be able decide to enable or disable the providers altogether no matter if they're beta or not.